sdk/python/kfp/v2/components/types/artifact_types.py

```diff
@@ -0,0 +1,451 @@
+# Copyright 2021 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Classes for input/output Artifacts in KFP SDK.
+
+These are only compatible with v2 Pipelines.
+"""
+
+import os
+from typing import Dict, Generic, List, Optional, Type, TypeVar, Union
+
+_GCS_LOCAL_MOUNT_PREFIX = '/gcs/'
+_MINIO_LOCAL_MOUNT_PREFIX = '/minio/'
+_S3_LOCAL_MOUNT_PREFIX = '/s3/'
+
+
+class Artifact(object):
+    """Generic Artifact class.
+
+  This class is meant to represent the metadata around an input or output
```

Comment:
nit: inconsistent indentation. Should we preserve 2-space indentation? or do you think we want to switch to 4-space indentation going forward?

Reply:
Yes, I'm thinking we should use 4-space going forward for consistency with external Google styles and PEP. WDYT?

Reply:
SGTM